tests/queries/0_stateless/02703_row_policy_for_database.sql

```diff
@@ -0,0 +1,53 @@
+-- Tags: no-parallel
+
+DROP DATABASE IF EXISTS db1_02703;
+DROP USER IF EXISTS u1_02703;
+CREATE USER u1_02703;
+
+CREATE DATABASE db1_02703;
+
+CREATE TABLE db1_02703.02703_rqtable (x UInt8) ENGINE = MergeTree ORDER BY x;
+INSERT INTO db1_02703.02703_rqtable VALUES (1), (2), (3), (4);
+
+
+SELECT '-- row policies for database';
+CREATE ROW POLICY db1_02703 ON db1_02703.* USING 1 AS PERMISSIVE TO ALL;
+CREATE ROW POLICY tbl1_02703 ON db1_02703.table USING 1 AS PERMISSIVE TO ALL;
+SELECT '  -- SHOW CREATE POLICY db1_02703 ON db1_02703.*';
+SHOW CREATE POLICY db1_02703 ON db1_02703.*;
+SELECT '  -- SHOW CREATE POLICY ON db1_02703.*';
+SHOW CREATE POLICY ON db1_02703.*;
+SELECT '  -- SHOW CREATE POLICY ON db1_02703.`*`';
```

Comment:
``` `*` ``` - is a valid table name, it shouldn't be used it as a marker.

I mean the following two lines should have different meanings:
```
CREATE ROW POLICY x ON y.`*` - should create a row policy on the table y.`*`
CREATE ROW POLICY x ON y.* - should create a row policy on the database y
```

I know `*` is quite a weird name, however it's allowed.

The same convention should work for SHOW CREATE ROW POLICY:
```
SHOW CREATE ROW POLICY ON y.`*` - should show row policies on the table y.`*`
SHOW CREATE ROW POLICY ON y.* - should create row policies on the database y and on the tables in this database
```

Reply:
It seems you can fix that just by replacing ANY_TABLE_MARK with something which can't be used as a table name (e.g. an empty string) and writing a bit more code.

Reply:
Hello Vitaly, thanks for your remarks.
Currently y.* shows all policies in database y (for all tables and database), while y.`*` shows policies for database y only.
Treating y.`*` as a  table name kind a limits functionality, although I totally agree with you that we have to respect `*` since it is a legit table name.
Will fix it.

Reply:
Please add a test with a row policy for a table named `*`. Probably it's better to do that in a separate test because it's a very special case.

Reply:
The test for `*` is already at the bottom of 02703_row_policies_for_database_combination.sql
Ok, I'll move it to the dedicated file.
I have to resolve a conflict in broken_tests.txt anyway.